grades/models.py

```diff
@@ -40,16 +40,20 @@ def course_level(self):
     def __unicode__(self):
         return self.code
     
-    @permalink
+    def __str__(self):
+        return self.code
+
+    # FIX Permalink depercated
     def get_absolute_url(self):
-        return 'course', None, {'course_code': self.code}
+        return reverse('course', kwargs={'course_code': self.code})
 
 
 class Grade(models.Model):
-    course = models.ForeignKey(Course)
+    course = models.ForeignKey(Course, on_delete=models.CASCADE)
     semester_code = models.CharField("Semester", max_length=10)
     
     average_grade = models.FloatField()
+    digitalExam = models.BooleanField(default=False)
```

Comment:
Should be `digital_exam` to keep up with Python naming practices.

Reply:
Yupp